### PR TITLE
refactor(eval_functions): Improve tests, docs, and readability

### PR DESCRIPTION
**Pull request by SandeepTuniki**
(Part of a series of general improvements I am doing with the help of Gemini CLI. These changes do not introduce any new functionality but make the existing code easier to understand, use, and maintain.)

This PR refactors and improves the `tools/statvar_importer/eval_functions.py` module and its corresponding tests.

**Key Changes:**

*   **Improved Testing:** The test suite in `eval_functions_test.py` is refactored for clarity by separating tests into distinct classes for each function. Test coverage is also improved by adding several new tests for edge cases and error conditions.
*   **Enhanced Documentation:** All docstrings in `eval_functions.py` are updated to improve clarity, along with added usage examples.

### CODE REVIEW

File: tools/statvar_importer/eval_functions.py [hunk 1]
```diff
@@ -97,20 +142,44 @@ def str_to_camel_case(input_string: str,
...
+        >>> # Statement without a variable
+        >>> evaluate_statement('2 * 2')
+        ('', 4)
+        >>> # Invalid expression
+        >>> evaluate_statement('name = 1 + "2"')
+        ('name', None)
+    """
     variable = ''
     statement = eval_str
     if '=' in eval_str:
```
Comment (ajaits) [inline tools/statvar_importer/eval_functions.py:185]: Can you extend this to catch any exception. It missed IndexError for some imports

**Review by ajaits — APPROVED** [review]

File: tools/statvar_importer/eval_functions.py (hunk 1 above)
Comment (SandeepTuniki) [reply tools/statvar_importer/eval_functions.py:185]: Done.
